test/cmdlineTests/standard_import_ast/input.json

```
@@ -0,0 +1,94 @@
+{
+	"language": "SolidityAST",
+	"sources": {
+		"A": {
+			"ast": {
+				"absolutePath": "A",
+				"exportedSymbols": {
+					"C": [
+						6
+					]
+				},
+				"id": 7,
+				"license": "GPL-3.0",
+				"nodeType": "SourceUnit",
+				"nodes": [
+					{
+						"id": 1,
+						"literals": [
+							"solidity",
+							">=",
+							"0.0"
+						],
+						"nodeType": "PragmaDirective",
+						"src": "36:22:0"
+					},
+					{
+						"abstract": false,
+						"baseContracts": [],
+						"canonicalName": "C",
+						"contractDependencies": [],
+						"contractKind": "contract",
+						"fullyImplemented": true,
+						"id": 6,
+						"linearizedBaseContracts": [
+							6
+						],
+						"name": "C",
+						"nameLocation": "68:1:0",
+						"nodeType": "ContractDefinition",
+						"nodes": [
+							{
+								"body": {
+									"id": 4,
+									"nodeType": "Block",
+									"src": "97:2:0",
+									"statements": []
+								},
+								"functionSelector": "26121ff0",
+								"id": 5,
+								"implemented": true,
+								"kind": "function",
+								"modifiers": [],
+								"name": "f",
+								"nameLocation": "81:1:0",
+								"nodeType": "FunctionDefinition",
+								"parameters": {
+									"id": 2,
+									"nodeType": "ParameterList",
+									"parameters": [],
+									"src": "82:2:0"
+								},
+								"returnParameters": {
+									"id": 3,
+									"nodeType": "ParameterList",
+									"parameters": [],
+									"src": "97:0:0"
+								},
+								"scope": 6,
+								"src": "72:27:0",
+								"stateMutability": "pure",
+								"virtual": false,
+								"visibility": "public"
+							}
+						],
+						"scope": 7,
+						"src": "59:42:0",
+						"usedErrors": []
+					}
+				],
+				"src": "36:65:0"
+			},
+			"id": 0
+		}
+	},
+	"settings": {
+		"outputSelection": {
+			"*": {
+				"": [
```

this looks strange for me.. but it seem to be defined like this

---

Maybe it's a consequence of this weird quirk we ran into in the asm import PR where using `""` as contract name in compiler stack would get you some contract and it of course happened to be the one you wanted if the stack had only one?

If that's it, then I guess it's not incorrect but this would be more in line with how we do it usually:

```suggestion
				"*": [
```

---

Yeah, it should really be ``"*"`` - not sure why ``""`` also works myself.

---

Hmm.. interesting.. I thought it just need to be like that - but I was not really checking other command-line-tests yet.. maybe that one is just a typo?

https://github.com/ethereum/solidity/blob/develop/test/cmdlineTests/standard_only_ast_requested/input.json#L14

---

Hm... even https://docs.soliditylang.org/en/v0.8.19/using-the-compiler.html documents this like ``""``...
Thinking about it for another moment, this is probably actually intentional.
The AST stretches over all sources anyways, so maybe this was intentionally removed from any specific source in the past and specifically required an empty source name.

---

So yeah, my earlier comment was wrong and this is actually correct and fine after all :-).

---

> The AST stretches over all sources anyways, so maybe this was intentionally removed from any specific source in the past and specifically required an empty source name.

ah great! that makes sense